### PR TITLE
Create an instance of the base program certificate for MicroMasters, XSeries and Professional Certificate

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
@edx/ecommerce

### CODE REVIEW

File: credentials/apps/credentials/views.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ class ExampleCredential(TemplateView):
     """ Example certificate. """
     template_name = 'credentials/base.html'
 
-    def get_context_data(self, **kwargs):
+    def get_context_data(self, program_type='MicroMasters', **kwargs):
```
Comment (rlucioni) [inline credentials/apps/credentials/views.py]: Like I said earlier about the catalog management command, we should be careful about hardcoding brand names in open-source code. If there's value in per-type example certificates, you should at least default to `None` here and use `Example` as the type when someone GETs `/credentials/example`.

Related: I understand what you're after here, but it doesn't seem wise to assume that whatever is in the path is a program type, then try to generate a certificate. By doing this, we open ourselves up to a whole class of errors we wouldn't otherwise have to worry about.

Comment (clintonb) [inline credentials/apps/credentials/views.py]: Don't modify the signature of this method. Use `kwargs` to pass values.

File: credentials/templates/credentials/images/micromasters-watermark.svg [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1847.86 2677.65" role="img" aria-hidden="true">
+<svg class="background-watermark" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1847.86 2677.65" role="img" aria-hidden="true">
```
Comment (clintonb) [inline credentials/templates/credentials/images/micromasters-watermark.svg]: I can foresee this causing issues if we ever use this image for another purpose. Rather than set this class here, set it where the image is actually used.

File: credentials/templates/credentials/program.html [hunk 3]
```diff
@@ -49,21 +49,15 @@ <h3 class="sr-only">{% trans "Print or share your certificate:" %}</h3>
 
         <main class="accomplishment accomplishment-main">
             <div class="wrapper-accomplishment-rendering">
-                <div class="accomplishment-rendering">
+                <div class="accomplishment-rendering" >
                     {% with certificate_context.program_details.type|slugify as program_type %}
                     {# http://stackoverflow.com/a/16656232 #}
```
Comment (clintonb) [inline credentials/templates/credentials/program.html]: This comment is not very useful. Please remove it.

File: credentials/templates/credentials/program.html [hunk 4]
```diff
@@ -49,21 +49,15 @@ <h3 class="sr-only">{% trans "Print or share your certificate:" %}</h3>
 
         <main class="accomplishment accomplishment-main">
             <div class="wrapper-accomplishment-rendering">
-                <div class="accomplishment-rendering">
+                <div class="accomplishment-rendering" >
```
Comment (clintonb) [inline credentials/templates/credentials/program.html]: This is unnecessary and should be reverted.

File: credentials/templates/credentials/program.html [hunk 5]
```diff
@@ -49,21 +49,15 @@ <h3 class="sr-only">{% trans "Print or share your certificate:" %}</h3>
...
                     {% with certificate_context.program_details.type|slugify as program_type %}
                     {# http://stackoverflow.com/a/16656232 #}
-                    <img class="watermark" src="{% static 'images/'|add:program_type|add:'-watermark.png' %}">
-                    {% endwith %}
+                    {% include 'credentials/images/'|add:program_type|add:'-watermark.svg' %}
```
Comment (clintonb) [inline credentials/templates/credentials/program.html]: Why are we including the raw SVG source rather than loading this as an image?

Comment (mikedikan) [inline credentials/templates/credentials/program.html]: Because we have to if we want the styles applied - this is a discussion that I had with @AlasdairSwan & @rlucioni and they can probably give the whole picture.

File: credentials/apps/credentials/tests/test_views.py [hunk 6]
```diff
@@ -179,3 +179,19 @@ def test_get(self):
...
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'micromasters-watermark')
+        self.assertContains(response, 'micromasters-logo')
+
+        response = self.client.get('/credentials/example/xseries')
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'xseries-watermark')
+        self.assertContains(response, 'xseries-logo')
+
+        response = self.client.get('/credentials/example/professional%20certificate')
```
Comment (MatthewPiatetsky) [inline credentials/apps/credentials/tests/test_views.py]: changed

File: credentials/templates/credentials/images/micromasters-watermark.svg (hunk 2 above)
Comment (MatthewPiatetsky) [inline credentials/templates/credentials/images/micromasters-watermark.svg]: changed

File: credentials/templates/credentials/program.html (hunk 4 above)
Comment (MatthewPiatetsky) [inline credentials/templates/credentials/program.html]: fixed

File: credentials/templates/credentials/program.html (hunk 3 above)
Comment (MatthewPiatetsky) [inline credentials/templates/credentials/program.html]: removed

File: credentials/apps/credentials/views.py (hunk 1 above)
Comment (MatthewPiatetsky) [inline credentials/apps/credentials/views.py]: @rlucioni I changed program_type to a url parameter. This way we don't make assumptions about it being a program unless the param is passed (and I default program_type to None). Does that work?

File: credentials/apps/credentials/views.py [hunk 7]
```diff
@@ -99,7 +100,7 @@ def get_context_data(self, **kwargs):
                 'program_details': ProgramDetails(
                     uuid=uuid.uuid4(),
                     title='Completely Example Program',
-                    type='Example',
+                    type=program_type,
```
Comment (rlucioni) [inline credentials/apps/credentials/views.py]: If you change this to
```py
type=program_type or 'Example',
```
You'll get a nicer looking example credential when no program type is specified.

File: credentials/templates/credentials/program.html [hunk 8]
```diff
@@ -51,19 +51,16 @@ <h3 class="sr-only">{% trans "Print or share your certificate:" %}</h3>
...
-                        <h2 class="accomplishment-title">
-                            {# TODO: Replace with SVG logo specific to program type. #}
-                            <span class="accomplishment-title-label">{% trans certificate_context.program_details.type %}</span>
-                            <span class="accomplishment-title-type">
-                                {% blocktrans with span_start='<span class="deco">' span_end='</span>' %}
-                                    Certificate {{ span_start }} of {{ span_end }} Achievement
-                                {% endblocktrans %}
-                            </span>
-                        </h2>
+                        {% if program_type != 'none' %}
```
Comment (rlucioni) [inline credentials/templates/credentials/program.html]: If you take my suggestion above, you can also get rid of this conditional.

Comment (MatthewPiatetsky) [inline credentials/templates/credentials/program.html]: but there is no watermark that looks like none-watermark or example-watermark. (same with logo)

File: credentials/apps/credentials/tests/test_views.py [hunk 9]
```diff
@@ -179,3 +179,19 @@ def test_get(self):
         """ Verify the view loads. """
         response = self.client.get(reverse('credentials:example'))
         self.assertEqual(response.status_code, 200)
+
+    def test_get_all_types(self):
```
Comment (rlucioni) [inline credentials/apps/credentials/tests/test_views.py]: `ddt` is your friend here.
```py
@ddt.data('micromasters', 'xseries', 'professional-certificate')
def test_get_example_credentials(self, program_type):
    response = self.client.get('{}?program_type={}'.format(reverse('credentials:example'), program_type)
    self.assertEqual(response.status_code, 200)
    for asset in ('watermark', 'logo'):
        self.assertContains(response, '{}-{}'.format(program_type, asset))
```

Also, this would be more complete with a test covering the empty case (when no program type is provided).

File: credentials/templates/credentials/program.html [hunk 10]
```diff
@@ -51,19 +51,16 @@ <h3 class="sr-only">{% trans "Print or share your certificate:" %}</h3>
...
-                        <h2 class="accomplishment-title">
-                            {# TODO: Replace with SVG logo specific to program type. #}
-                            <span class="accomplishment-title-label">{% trans certificate_context.program_details.type %}</span>
-                            <span class="accomplishment-title-type">
-                                {% blocktrans with span_start='<span class="deco">' span_end='</span>' %}
-                                    Certificate {{ span_start }} of {{ span_end }} Achievement
-                                {% endblocktrans %}
-                            </span>
-                        </h2>
+                        {% if program_type != 'example' %}
```
Comment (rlucioni) [inline credentials/templates/credentials/program.html]: Why is this necessary? Consider adding an `example-` watermark and logo so that the basic example credential looks good.

File: credentials/apps/credentials/tests/test_views.py (hunk 9 above)
Comment (MatthewPiatetsky) [inline credentials/apps/credentials/tests/test_views.py]: thanks, changed. didn't do this previously when i had a space in professional certificate but now this works

File: credentials/templates/credentials/program.html (hunk 10 above)
Comment (MatthewPiatetsky) [inline credentials/templates/credentials/program.html]: added example watermark and logo

File: credentials/templates/credentials/program.html (hunk 5 above)
Comment (rlucioni) [inline credentials/templates/credentials/program.html]: @clintonb our commitment to supporting IE11 prevents us from using more modern methods of styling SVGs (e.g., CSS masks). An SVG included via CSS as a background image can't be easily styled without these. For more, see https://codepen.io/noahblon/post/coloring-svgs-in-css-background-images.
